lib/libutee/include/utee_defines.h

```diff
@@ -168,6 +168,70 @@ typedef enum {
 #define TEE_MAC_SIZE_AES_CMAC
 #define TEE_MAC_SIZE_DES_CBC_MAC_PKCS5
 
+static inline size_t __tee_hash_get_digest_size(uint32_t algo)
+{
+	switch (algo) {
+	case TEE_ALG_MD5:
+	case TEE_ALG_HMAC_MD5:
+		return TEE_MD5_HASH_SIZE;
+	case TEE_ALG_SHA1:
+	case TEE_ALG_HMAC_SHA1:
+	case TEE_ALG_DSA_SHA1:
+		return TEE_SHA1_HASH_SIZE;
+	case TEE_ALG_SHA224:
+	case TEE_ALG_HMAC_SHA224:
+	case TEE_ALG_DSA_SHA224:
+		return TEE_SHA224_HASH_SIZE;
+	case TEE_ALG_SHA256:
+	case TEE_ALG_HMAC_SHA256:
+	case TEE_ALG_DSA_SHA256:
+		return TEE_SHA256_HASH_SIZE;
+	case TEE_ALG_SHA384:
+	case TEE_ALG_HMAC_SHA384:
+		return TEE_SHA384_HASH_SIZE;
+	case TEE_ALG_SHA512:
+	case TEE_ALG_HMAC_SHA512:
+		return TEE_SHA512_HASH_SIZE;
+	case TEE_ALG_SM3:
+	case TEE_ALG_HMAC_SM3:
+		return TEE_SM3_HASH_SIZE;
+	default:
+		return 0;
+	}
+}
+
+static inline size_t __tee_mac_get_digest_size(uint32_t algo)
```

Comment:
I think it would be enough with one function, let's say `__tee_get_digest_size()`. The caller can use the `TEE_ALG_GET_CLASS()` macro first if only one class should be used.

I'm not sure if we should use this pattern with an inline function together with a macro or not. I'm leaning towards it, but what does others think? @jforissier @etienne-lms @jbech-linaro

Reply:
Well it seems good for consistency with existing code at least.

Reply:
I really wonder if an inline function is really useful for that. As more algo come, the helper grows a bit. Is there really a need for performance when this helper is used? All in one, I would be in favor of a standard function `tee_alg_get_digest_size(uint32_t algo)`. Not a strong opinion yet.

If going for a macro `TEE_ALG_GET_DIGEST_SIZE()` then, as @jforissier said about consistency with other helpers above, the macro referring to an inline is fine.

Reply:
An disadvantage with a standard function which would end up in a C file instead is that OP-TEE Core doesn't link with libutee so we'd need something extra. Performance isn't an issue and so isn't code size either since this is used very sparingly.

Reply:
ok, so macro/inline is fine to me.

Reply:
Alright, then I will try to make it work with just a single function, keeping the inline function + macro pattern